dev/TeachingTip/APITests/TeachingTipTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Markup;
+using MUXControlsTestApp.Utilities;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media.Imaging;
+using Common;
+using System.Threading;
+
+
+#if USING_TAEF
+using WEX.TestExecution;
+using WEX.TestExecution.Markup;
+using WEX.Logging.Interop;
+#else
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.Logging;
+#endif
+
+#if !BUILD_WINDOWS
+using TeachingTip = Microsoft.UI.Xaml.Controls.TeachingTip;
+#endif
+
+namespace Windows.UI.Xaml.Tests.MUXControls.ApiTests
+{
+    [TestClass]
+    public class TeachingTipTests
+    {
+        [TestMethod]
+        public void TeachingTipBackgroundTest()
+        {
+            var resetEvent = new AutoResetEvent(false);
+            RunOnUIThread.Execute(() =>
+            {
+                TeachingTip teachingTip = new TeachingTip();
+                teachingTip.Loaded += (object sender, RoutedEventArgs args) => { resetEvent.Set(); };
+                MUXControlsTestApp.App.TestContentRoot = teachingTip;
+            });
+
+            IdleSynchronizer.Wait();
+            resetEvent.WaitOne();
+
+            RunOnUIThread.Execute(() =>
+            {
+                TeachingTip teachingTip = (TeachingTip)MUXControlsTestApp.App.TestContentRoot;
+                var redBrush = new SolidColorBrush(Colors.Red);
+                teachingTip.SetValue(TeachingTip.BackgroundProperty, redBrush);
+                Verify.AreSame(redBrush, teachingTip.GetValue(TeachingTip.BackgroundProperty) as Brush);
+                Verify.AreSame(redBrush, teachingTip.Background);
+
+                var blueBrush = new SolidColorBrush(Colors.Blue);
+                teachingTip.Background = blueBrush;
+                Verify.AreSame(blueBrush, teachingTip.Background);
+
+                var child = VisualTreeHelper.GetChild(teachingTip, 0);
+                var grandChild = VisualTreeHelper.GetChild(child, 1);
+                Verify.AreSame(blueBrush, ((Grid)grandChild).Background);
+
+                teachingTip.IsLightDismissEnabled = true;
+            });
+
+            IdleSynchronizer.Wait();
+
+            RunOnUIThread.Execute(() =>
+            {
+                TeachingTip teachingTip = (TeachingTip)MUXControlsTestApp.App.TestContentRoot;
+                var blueBrush = new SolidColorBrush(Colors.Blue);
+                Verify.AreNotEqual(blueBrush, teachingTip.Background);
```

Comment:
I don't think this test is validating much. Since you are creating a new Brush, there would be no possibility of the teachingTip using it.

If you want to compare, you should compare the Color values of the brushes.

Also: why would the Background property not be Blue in this scenario? Given that you have just set in to blue in the previous block?

Reply:
You are right the background property should not change this test was to make sure that setting the IsLightDismissEnabled to true changes the background of the content of the tip to an acrylic brush, or fallback.

---
In reply to: [252901760](https://github.com/Microsoft/microsoft-ui-xaml/pull/32#discussion_r252901760) [](ancestors = 252901760)